test/stmt/switch_nil.swift

```diff
@@ -0,0 +1,15 @@
+// RUN: %target-typecheck-verify-swift
+
+enum Hey {
+  case listen
+}
+
+func test() {
+  switch Hey.listen {
```

Comment:
Could you please add a test-case where this switch is used in a result builder body? Pattern matching is also handled by the solver now so we need to make sure that behavior is consistent…

Reply:
Good idea.